### PR TITLE
Update wait command docs

### PR DESCRIPTION
**Pull request by kyleknap**
Docs were updated to include how timeouts happen. There has been some confusion from users on how waiter timeouts. Here is an example of how it would look for ``s3api wait bucket-exists``:

```
NAME
       bucket-exists -

DESCRIPTION
       Wait  until  200 response is received when polling with head-bucket. It
       will poll every 5 seconds until a successful state has been reached. An
       error is thrown after 20 failed checks.
```

cc @jamesls

### CODE REVIEW

File: awscli/customizations/waiters.py [hunk 1]
```diff
@@ -181,6 +183,13 @@ def _build_operation_description(self, operation):
         operation_name = xform_name(operation).replace('_', '-')
         return u'when polling with ``%s``.' % operation_name
 
+    def _build_polling_description(self, delay, max_attempts):
+        description = (
+            ' It will poll every %s seconds until a successful state '
+            'has been reached. An error is thrown after %s failed checks.'
```
Comment (jamesls) [inline awscli/customizations/waiters.py]: Can we reword this to be more cli-centric.  This will exit with an RC of <some rc> after xx many failed checks.  The CLI doesn't really throw exceptions.

**Comment by jamesls** [comment]
:shipit: Just had one small comment.

File: awscli/customizations/waiters.py (hunk 1 above)
Comment (kyleknap) [inline awscli/customizations/waiters.py]: sounds good

**Comment by kyleknap** [comment]
Updated PR to mention the return code. Merging.
